libc/string/string_cmp.c

```diff
@@ -0,0 +1,544 @@
+/*
+ * Phoenix-RTOS
+ *
+ * POSIX.1-2017 standard library functions tests
+ * HEADER:
+ *    - string.h
+ * TESTED:
+ *    - memcmp()
+ *    - strcmp()
+ *    - strncmp()
+ *    - strcoll()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Damian Modzelewski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <unity_fixture.h>
+
+#include "testdata.h"
+
+#define BUFF_SIZE 129
+#define BIG_SIZE  1024
+
+static char empty[BUFF_SIZE];
+
+TEST_GROUP(string_memcmp);
+TEST_GROUP(string_strncmp);
+TEST_GROUP(string_strcmp);
+TEST_GROUP(string_strcoll);
+
+
+/*
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+*/
+
+
+TEST_SETUP(string_memcmp)
+{
+}
+
+
+TEST_TEAR_DOWN(string_memcmp)
+{
+}
+
+
+TEST(string_memcmp, basic)
+{
+	const char *basicStr = "Test";
+
+	TEST_ASSERT_EQUAL_INT(0, memcmp(basicStr, basicStr, 5));
+	TEST_ASSERT_LESS_THAN_INT(0, memcmp(basicStr, "Tests", 5));
+	TEST_ASSERT_GREATER_THAN_INT(0, memcmp("Tests", basicStr, 5));
+}
+
+
+TEST(string_memcmp, unsigned_char_cast)
+{
+	const signed char charEdgeValue = SCHAR_MIN;
+
+	TEST_ASSERT_GREATER_THAN_INT(0, memcmp(&charEdgeValue, "\0", 1));
+	TEST_ASSERT_LESS_THAN_INT(0, memcmp("\0", &charEdgeValue, 1));
+}
+
+
+TEST(string_memcmp, emptyInput)
+{
+	char *asciiSet = testdata_createCharStr(BUFF_SIZE),
+		 separated[24] = "\0\0\0\0\0TEST\0\0";
```

Comment:
IMHO we should choose one of the following options, because the current one may be misleading:

 - separated[] = "\0\0\0\0\0TEST\0\0\0\0\0\0\0\0\0\0\0\0\0\0"; // sttring with size 24 for init
 - separated[] = "\0\0\0\0\0TEST\0\0"; //current sttring for init
 
I think the second one would be sufficient.